modules/fedimint-ln-client/src/pay.rs

```diff
@@ -258,6 +294,29 @@ impl LightningPayFunded {
     async fn gateway_pay_invoice(
         gateway: LightningGateway,
         payload: PayInvoicePayload,
+    ) -> Result<String, GatewayPayError> {
+        // Abort the payment if we can't reach the gateway within 30 seconds
+        // to prevent unexpected delays for the user.
+        let deadline = fedimint_core::time::now() + GATEWAY_API_TIMEOUT;
+
+        let mut last_error = None;
+        while fedimint_core::time::now() < deadline {
+            match Self::try_gateway_pay_invoice(gateway.clone(), payload.clone()).await {
```

Comment:
Hmm it's probably fine for now, but it looks like this will retry even if the gateway returns a 5XX error. This means that the gateway is reachable, but something went wrong when processing the payment. I don't think we should retry in that case.

Reply:
I think it's reasonable to retry even on 5xx errors. For instance you may be getting 502 Bad Gateway because the server is down or a 500 Internal Server Error may happen because the LN node is behaving in a weird way. But of course in the long term we should try to get more granular on these different errors.